File: Array/LC_49.cpp
// Written by: Jian-Yuan Lin
// Date: 2022.11.17

class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> map;

        for (string str: strs){
            string s = str;
            sort(s.begin(), s.end());
            map[s].push_back(str);
        }
        vector<vector<string>> ans;
        for (auto it = map.begin(); it != map.end(); it++){
            ans.push_back(it->second);
        }
        return ans;
    }
};

File: Array/LC_49.py
# Written by: Jian-Yuan Lin
# Date: 2022.11.17

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        mapping = collections.defaultdict(list)
        for item in strs:
            sorted_string = "".join(sorted(item))
            mapping[sorted_string].append(item)
        
        ans = []
        for item in mapping:
            ans.append(mapping[item])
        return ans